pkg/vmcp/server/session_adapter.go

```diff
@@ -0,0 +1,148 @@
+package server
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+
+	"github.com/stacklok/toolhive/pkg/logger"
+	"github.com/stacklok/toolhive/pkg/transport/session"
+)
+
+// sessionIDAdapter adapts ToolHive's session.Manager to implement
+// the mark3labs SDK's SessionIdManager interface.
+//
+// Session Management Ownership:
+//   - ALL session storage, TTL, and cleanup is handled by ToolHive's session.Manager
+//   - The mark3labs SDK does NOT manage sessions - it only calls our interface
+//   - This is the SAME session.Manager used by pkg/transport/proxy/streamable for MCP sessions
+//
+// The SDK calls our adapter methods during MCP protocol flows:
+//  1. Generate() - When client sends MCP initialize without Mcp-Session-Id header
+//  2. Validate() - On every request to check if session exists and is active
+//  3. Terminate() - When client sends HTTP DELETE to end session
+//
+// All three methods delegate to ToolHive's session.Manager. The adapter is purely
+// infrastructure glue code (not domain logic) and correctly lives within the
+// server bounded context per DDD principles.
+type sessionIDAdapter struct {
+	manager *session.Manager
+}
+
+// newSessionIDAdapter creates an adapter that bridges session.Manager
+// to the mark3labs SDK's SessionIdManager interface.
+func newSessionIDAdapter(manager *session.Manager) *sessionIDAdapter {
+	return &sessionIDAdapter{
+		manager: manager,
+	}
+}
+
+// Generate creates a new session ID and registers it with the session manager.
+// Called by the SDK when handling MCP initialize requests without a session ID.
+//
+// Per MCP spec (https://modelcontextprotocol.io/specification/2025-06-18/basic/transports):
+//   - Session ID SHOULD be globally unique and cryptographically secure
+//   - Session ID MUST only contain visible ASCII characters (0x21 to 0x7E)
+//   - Server returns session ID in Mcp-Session-Id header on InitializeResult
+func (a *sessionIDAdapter) Generate() string {
+	// Generate a cryptographically secure UUID as per MCP spec requirements
+	sessionID := uuid.New().String()
+
+	// Create and register the session with the manager
+	if err := a.manager.AddWithID(sessionID); err != nil {
+		// This shouldn't happen unless we have a UUID collision (extremely unlikely)
+		logger.Errorf("Failed to create session %s: %v", sessionID, err)
+		// Generate another ID and try once more
+		sessionID = uuid.New().String()
+		if err := a.manager.AddWithID(sessionID); err != nil {
+			// Log but return the ID anyway - the SDK will use it
+			// If storage is truly broken, validation will catch it
+			logger.Errorf("Failed to create session %s on retry: %v", sessionID, err)
```

Comment:
we are erroring here but still continuing the flow, adding a second debug message and returning session id?